pypeit/calibrations.py

```diff
@@ -560,40 +562,55 @@ def get_flats(self):
-
-            # If we tweaked the slits, update the tilts_dict and
-            # tslits_dict to reflect new slit edges
-            if self.par['flatfield']['tweak_slits']:
-                msgs.info('Using slit boundary tweaks from IllumFlat and updated tilts image')
-                self.tslits_dict = self.flatField.tslits_dict
-                self.tilts_dict = self.flatField.tilts_dict
+            self.mspixelflat, self.msillumflat = self.flatField.run(show=self.show)
+
+            # TODO: Tilts are unchanged, right?
+#            # If we tweaked the slits, update the tilts_dict and
+#            # tslits_dict to reflect new slit edges
+#            if self.par['flatfield']['tweak_slits']:
+#                # flatfield updates slits directly and only alters the
+#                # *_tweak set. This updates the MasterEdges
+#                msgs.info('Using slit boundary tweaks from IllumFlat and updated tilts image')
+##                self.tslits_dict = self.flatField.tslits_dict
+#                self.tilts_dict = self.flatField.tilts_dict
 
             # Save to Masters
             if self.save_masters:
                 self.flatField.save()
 
-                # If we tweaked the slits update the master files for tilts and slits
-                # TODO: These should be saved separately
+                # If slits were tweaked by the slit illumination
+                # profile, re-write them so that the tweaked slits are
+                # included.
                 if self.par['flatfield']['tweak_slits']:
-                    msgs.info('Updating MasterTrace and MasterTilts using tweaked slit boundaries')
-                    self.edges.update_using_tslits_dict(self.flatField.tslits_dict)
-                    self.edges.save()
-                    # Write the final_tilts using the new slit boundaries to the MasterTilts file
-                    self.waveTilts.final_tilts = self.flatField.tilts_dict['tilts']
-                    self.waveTilts.tilts_dict = self.flatField.tilts_dict
-                    self.waveTilts.save()
+                    self.slits.to_master()
+
+#                # If we tweaked the slits update the master files for tilts and slits
+#                # TODO: These should be saved separately
+#                if self.par['flatfield']['tweak_slits']:
+#                    msgs.info('Updating MasterTrace and MasterTilts using tweaked slit boundaries')
+#                    # flatfield updates slits directly and only alters
+#                    # the *_tweak set. This updates the MasterEdges
+#                    # file with the new data from the flat-field slit
+#                    # tweaks.
+#                    # TODO: Make SlitTraceSet a masterframe?
+#                    self.edges.update_slits(self.slits)
+#                    self.edges.save()
+
+                    # TODO: Tilts are unchanged, right?
```

Comment:
I think I see now.  Thanks for catching this.  As far as I can tell, `tilts` itself doesn't change because it's extrapolated over the full image for each slit.  It's the mask that isolates which pixels are part of the slit that changes, and that's what actually makes `final_tilts` different from the input tilts in `flatfield.run` (in the current `develop` branch).  I'll get this fixed.

Reply:
That is right, the tilts don't need to be re-fit after a tweak because the tilt image independent of the exact location of the boundaries.

Reply:
This is fixed now in `FlatField.fit`.  Look for `tweaked_tilts`.